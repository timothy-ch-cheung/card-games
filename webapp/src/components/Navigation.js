import React from "react";
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar'
import '../App.css';
import {NavLink} from "react-bootstrap";
import {LinkContainer} from 'react-router-bootstrap'
import {useSelector, useDispatch} from "react-redux";

class Navigation extends React.Component {

    render() {
        return (
            <Navbar bg="dark" variant="dark" expand="lg">
                <LinkContainer to="/">
                    <Navbar.Brand>Noughts and Crosses</Navbar.Brand>
                </LinkContainer>
                <Navbar.Toggle aria-controls="basic-navbar-nav"/>
                <Navbar.Collapse>
                    <Nav className="mr-auto">
                        <LinkContainer to="/home"><NavLink>Home</NavLink></LinkContainer>
                        <LinkContainer to="/games"><NavLink>Games</NavLink></LinkContainer>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        );
    }
}

export default Navigation;