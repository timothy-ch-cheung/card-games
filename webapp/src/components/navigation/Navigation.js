import React from "react";
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar'
import '../../App.css';
import {NavLink} from "react-bootstrap";
import {LinkContainer} from 'react-router-bootstrap'
import Image from "react-bootstrap/Image";

class Navigation extends React.Component {

    render() {
        return (
            <Navbar bg="dark" variant="dark" expand="lg">
                <LinkContainer to="/">
                    <Navbar.Brand className={"no-drag"}>
                        <Image
                            src="logo.svg"
                            width="32"
                            height="32"
                            className="d-inline-block align-top no-drag"
                        />
                        {' '}Card Games</Navbar.Brand>
                </LinkContainer>
                <Navbar.Toggle aria-controls="basic-navbar-nav"/>
                <Navbar.Collapse>
                    <Nav className="mr-auto">
                        <LinkContainer to="/home"><NavLink className={"no-drag"}>Home</NavLink></LinkContainer>
                        <LinkContainer to="/games"><NavLink className={"no-drag"}>Games</NavLink></LinkContainer>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        );
    }
}

export default Navigation;