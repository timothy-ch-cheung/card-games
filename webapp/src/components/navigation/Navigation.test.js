import React from 'react';
import renderer from "react-test-renderer";
import Navigation from "./Navigation";
import {MemoryRouter} from "react-router-dom";

test('matches snapshot', () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={['/']}><
            Navigation/>
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot()
});