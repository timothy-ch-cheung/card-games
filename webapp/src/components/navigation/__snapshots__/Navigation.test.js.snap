// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="active navbar-brand"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Noughts and Crosses
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="/home"
        href="/home"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/games"
        href="/games"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Games
      </a>
    </div>
  </div>
</nav>
`;
